### ISSUE TITLE
Auto restart the XMPP connection if there was a settings change that affects only new connections

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#190]
```
when i login with gmail username and password, the app show invalid username or 
password...
GTalkSMS version = 2.24
Android OS version = 2.2
Phone model = samsung tab 
```

Original issue reported on code.google.com by `sherry.x...@gmail.com` on 20 Jun 2011 at 5:01

**Comment by GoogleCodeExporter** [comment]
```
Did you double check your login password ? 
Do you have some spaces at the end ? 
Do you add the server extension to your login (ex: toto@jabber.org) ?
Are you using xmpp or gtalk ? with one or two accounts ?
The check box "manually configure server" is it checked ?
```

Original comment by `Florent....@gmail.com` on 29 Jun 2011 at 6:24
- Changed title: **Invalid username or password error**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 19 Jul 2011 at 7:06
- Changed state: **UserFeedBack**

**Comment by GoogleCodeExporter** [comment]
```
I noticed, that you have to stop the service, then close the app for changes to 
take effect. I always checked if it was really closed in a Taskmanager and 
killed it before restarting if I saw it (I don't know if that was necessary). 
Maybe gtalksms should auto disconnect and reload settings after returning from 
settings? Or make this checkable in Settings. 
```

Original comment by `marx.hen...@googlemail.com` on 19 Aug 2011 at 6:40

**Comment by GoogleCodeExporter** [comment]
```
We'll think about that. Until then: Pressing the Start/Stop button should also 
force the new settings on the following connection.
```

Original comment by `fschm...@gmail.com` on 19 Aug 2011 at 8:13

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 25 Aug 2011 at 10:20
- Changed title: **Auto restart the XMPP connection if there was a settings change that affects only to new connections**
- Changed state: **New**
- Added labels: **Priority-Low**, **Type-Enhancement**
- Removed labels: **Priority-Medium**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 26 Aug 2011 at 4:11
- Changed title: **Auto restart the XMPP connection if there was a settings change that affects only new connections**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 1 Dec 2011 at 5:36
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Working on a new user interface that will fix these issues
```

Original comment by `Florent....@gmail.com` on 20 May 2012 at 12:09
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
The mechanism is already in place
```

Original comment by `Florent....@gmail.com` on 10 Jun 2012 at 3:58
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed with GTalkSMS v4.0 and released on market
```

Original comment by `Florent....@gmail.com` on 28 Jul 2012 at 4:20
- Changed state: **Deployed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed with GTalkSMS v4.0 and released on market
```

Original comment by `Florent....@gmail.com` on 28 Jul 2012 at 4:20
